ci/build_conda.sh

```diff
@@ -15,6 +15,16 @@ rapids-print-env
 
 rapids-logger "Begin py build"
 
-rapids-conda-retry mambabuild conda/recipes/pynvjitlink
+export CUDA_VERSION="$(cat pynvjitlink/CUDA_VERSION)"
+
+cat > cuda_compiler_version.yaml << EOF
+cuda_compiler_version:
+  - "${CUDA_VERSION}"
+EOF
```

Comment:
As `cuda_compiler_version` isn't captured in the hash unless it is in a variant file, this writes out a custom variant

In this case it looks like

```yaml
cuda_compiler_version:
  - "12.6"
```